GDS-to-PNG rendering: Remove labels, improve compression

algofoogle opened the pull request:
Changes how the `gds` GHA renders the GDS to PNG, to greatly reduce the file size (hoping to get under 5MB for an 8x2 tile).

For info on what this does, see this older PR: https://github.com/TinyTapeout/tt03p5-submission-template/pull/4

NOTE: This uses `pngquant --quality 10-30 ...` to try and get an 8x2 tile to render to a small enough PNG file size, but it does mean some reduction in fidelity... but for most this shouldn't be noticeable. The quality option can maybe be removed conditionally in a future revision (e.g. for smaller tiles), but it might be better to work on integrating parts of this overall approach into `tt_tool.py --create-png` instead.

algofoogle committed:
GDS-to-PNG rendering: Remove labels, improve compression

urish commented on action.yml:
```diff
@@ -83,9 +90,20 @@ runs:
+      shell: bash
+      run: |
+        pngquant --quality 10-30 --speed 1 --nofs --strip --output gds_render.png gds_render_png24.png
+        ls -al gds_render*.png
```
Any reason to keep this `ls -al` here?

algofoogle replied:
Probably don't need to keep it. It was in there as a way to show the size difference between original PNG and quantised (recompressed) PNG, mostly for diagnostics.

urish replied:
Alright, can you please remove it then?

algofoogle replied:
OK, removed in https://github.com/TinyTapeout/tt-gds-action/pull/1/commits/bae56eec67d05458b812492680366a35f5617cf3

urish commented on action.yml:
```diff
@@ -83,9 +90,20 @@ runs:
+    - name: Convert SVG to PNG
       shell: bash
-      run: ./tt/tt_tool.py --create-png
+      run: rsvg-convert --unlimited gds_render.svg -s <(echo 'text{display:none;}') -o gds_render_png24.png --no-keep-image-data
```
What's the purpose of `<(echo 'text{display:none;}')` ?

algofoogle replied:
That hides text labels when rendering the PNG (which otherwise become a cluttered mess). `rsvg-convert`'s `-s` parameter expects a CSS file as input; this inlines it so a separate file isn't needed.

urish replied:
Thanks for explaining! Can you please add a comment to the yaml that explain this?

algofoogle replied:
OK, comment added in https://github.com/TinyTapeout/tt-gds-action/pull/1/commits/bae56eec67d05458b812492680366a35f5617cf3

algofoogle committed:
GDS-to-PNG rendering: Minor tidy-up

algofoogle commented:
Commit https://github.com/TinyTapeout/tt-gds-action/pull/1/commits/bae56eec67d05458b812492680366a35f5617cf3 added: Removed the `ls` and added the comment explaining the inline CSS.

urish commented:
Lovely, thanks!

urish commented:
Just tested it with [TinyTapeout/tt04-usbcdc](https://github.com/TinyTapeout/tt04-usbcdc/actions/runs/5369552034). First of all, I can finally see the PNG (as it's less than 5MB). And it's not cluttered at all! 

https://github.com/TinyTapeout/tt04-usbcdc/actions/runs/5369552034

![image](https://github.com/TinyTapeout/tt-gds-action/assets/892318/fa920460-76e2-4ad6-9b20-d110c94d665c)
